src/backend/distributed/executor/distributed_intermediate_results.c

```
@@ -411,9 +411,16 @@ ExecuteSelectTasksIntoTupleStore(List *taskList, TupleDesc resultDescriptor,
 	Tuplestorestate *resultStore = tuplestore_begin_heap(randomAccess, interTransactions,
 														 work_mem);
 
+	/*
+	 * Local execution is not supported because here we use perPlacementQueryStrings.
+	 * Local execution does not know how to handle it. One solution is to extract and set
+	 * queryStringLazy from perPlacementQueryStrings. The extracted one should be the
+	 * query string for the local placement.
+	 */
```

Going ahead with the INSERT..SELECT is potentially dangerous if we already modified one of the shard placements via local execution. Should we throw an error here? (or fix it somehow)

---

If we used local execution and this cannot be executed locally but accesses a local shard then error will be thrown here: https://github.com/citusdata/citus/pull/3636/files#diff-7f83cda7e22ea5663ea713e1a36fcb29R975